client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.controller.js

```diff
@@ -1,56 +1,68 @@
-    }
-
-    $scope.deleteRenew = function () {
-        const AUTO_RENEW_TYPES = ["automaticV2014", "automaticV2016", "automaticForcedProduct"];
-        const renewType = $scope.selectedServices[0].service ? $scope.selectedServices[0].service.renewalType : $scope.selectedServices[0].renewalType;
-        if ($scope.selectedServices[0].renew.automatic && !AUTO_RENEW_TYPES.includes(renewType)) {
-            setManual()
-                .then(setDelete)
-                .catch(errorFunction)
-                .finally(() => {
-                    $rootScope.$broadcast(BillingAutoRenew.events.AUTORENEW_CHANGED);
-                    $scope.resetAction();
+angular
+    .module("Billing.controllers")
+    .controller("billingAutoRenewDeleteCtrl", class BillingAutoRenewDeleteCtrl {
+        constructor ($filter, $q, $rootScope, $scope, $translate, Alerter, BillingAutoRenew, AUTORENEW_EVENT) {
+            this.$filter = $filter;
+            this.$q = $q;
+            this.$rootScope = $rootScope;
+            this.$scope = $scope;
+            this.$translate = $translate;
+
+            this.Alerter = Alerter;
+            this.BillingAutoRenew = BillingAutoRenew;
+
+            this.AUTORENEW_EVENT = AUTORENEW_EVENT;
+        }
+
+        $onInit () {
+            this.selectedServices = this.$scope.currentActionData;
+            this.serviceToDisplay = this.selectedServices[0];
+            this.serviceToDisplay.expirationText = this.$filter("date")(this.serviceToDisplay.expiration, "mediumDate");
+        }
+
+        switchRenewalModeToManual () {
+            const formattedServices = _(this.selectedServices).map((originalService) => {
+                const service = _(originalService).clone(true);
+                service.renew.automatic = false;
+
+                return _(service).pick(["serviceId", "serviceType", "renew"]).value();
+            }).value();
+
+            return this.BillingAutoRenew.updateServices(formattedServices);
+        }
+
+        switchRenewalModeToDeleteAtExpiration () {
+            const formattedServices = _(this.selectedServices).map((originalService) => {
+                const service = _(originalService).clone(true);
+                service.renew.deleteAtExpiration = true;
+
+                return _(service).pick(["serviceId", "serviceType", "renew"]).value();
+            }).value();
+
+            return this.BillingAutoRenew
+                .updateServices(formattedServices)
+                .then(() => {
+                    this.$scope.$emit(this.AUTORENEW_EVENT.TERMINATE_AT_EXPIRATION, formattedServices);
                 });
-        } else {
-            setDelete()
-                .catch(errorFunction)
+        }
+
+        switchRenewalModeToManualIfNeeded () {
+            const AUTO_RENEW_TYPES = ["automaticV2014", "automaticV2016", "automaticForcedProduct"];
+            const renewType = _(this.serviceToDisplay).get("service.renewalType") || this.serviceToDisplay.renewalType;
```

Comment:
Why not write directly `_(this.serviceToDisplay).get("service.renewalType",this.serviceToDisplay.renewalType)` instead ? 😄